### PR TITLE
updated form validation and error scenarios

### PR DESCRIPTION
**Pull request by joeyymoore**
### Description
Updated form validation and error scenarios/messages

Add a link to the relevant story in Jira
https://jira.tools.tax.service.gov.uk/browse/SASS-1055

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [x]  Have you checked to ensure all dependencies are up to date?
- [x]  Have you checked to ensure its been rebased against the current version of master?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [x]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [x]  Have you checked for merge conflicts or any changes in the current master that may affect the current pull request? i.e. does it need another rebase?
- [x]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [x]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [ ]  Have you run the tests?
- [ ]  Have you run the journey tests? (where applicable)
- [ ]  Have you addressed warnings where appropriate?
- [ ]  Have you rebased against the current version of master?
- [ ]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [ ]  Have you checked the PR Builder passes?

### CODE REVIEW

File: app/forms/employment/EmploymentStartDateForm.scala [hunk 1]
```diff
@@ -35,34 +30,47 @@ object EmploymentStartDateForm {
...
+      case (true, false, false) => Seq(FormError("emptyDay", s"employment.employmentStartDate.error.incompleteDay.$agentCheck"))
+      case (false, false, false) => Seq()
+    }
+  }
+
+  def verifyNewDate(date: EmploymentDate, taxYear: Int, isAgent: Boolean): Seq[FormError] = {
+    val agentCheck = if (isAgent) "agent" else "individual"
+    val areDatesEmptyCall = areDatesEmpty(date, isAgent)
+    areDatesEmptyCall.isEmpty match {
+      case true =>
```
Comment (ina-tu) [inline app/forms/employment/EmploymentStartDateForm.scala:62]: where areDateEmptyCall.isEmpty is true it reads as if it should be false. I'm not sure if the name should be changed as the logic makes sense

**Review by ina-tu — CHANGES_REQUESTED** [review]
Dependancies can be updated:
Screenshot 2021-08-11 at 10 13 43

**Review by ina-tu — COMMENTED** [review]
branch needs to be rebased with master

File: app/forms/employment/EmploymentStartDateForm.scala (hunk 1 above)
Comment (joeyymoore) [reply app/forms/employment/EmploymentStartDateForm.scala]: renamed to hopefully make it more readable

**Review by ina-tu — APPROVED** [review]

**Pull request merged**